rfc/multiple-go-modules.md

```diff
@@ -0,0 +1,187 @@
+# [RFC] Split VMClarity into multiple go modules
+
+*Note: this RFC template follows HashiCrop RFC format described [here](https://works.hashicorp.com/articles/rfc-template)*
+
+|               |                                             |
+| ------------- | ------------------------------------------- |
+| **Created**   | 2024-01-17                                  |
+| **Status**    | **WIP** \| InReview \| Approved \| Obsolete |
+| **Owner**     | paralta                                     |
+| **Approvers** | vmclarity-maintainers                       |
+
+---
+
+This RFC proposes a new project structure which splits the main module into multiple modules to allow us to version modules separately and reduce dependencies.
+
+## Background
+
+VMClarity consists of a single module containing all the packages required to run VMClarity. However, there is a clear boundary between the following packages: api, cli, orchestrator, ui and uibackend.
+
+The build and deployment of these packages is already performed separately and independently.
+
+## Proposal
+
+The proposal here is to split the VMClarity repository into multiple modules:
+
+- **api**. Interface between all the services in VMClarity including the DB. Composed by API model, backend client and server.
+- **cli**. Responsible for running a scan in an asset and report the results back to api. Contains the logic to configure, run and manage different analysers and scanners. 
```

Comment:
does this mean that the cli takes ownership of scanning capabilities?

Reply:
Yes, it does. And at some point we might want to remove the scanner from the CLI, we can address this on the scope of https://github.com/openclarity/vmclarity/issues/346.